pycparser/c_generator.py

```diff
@@ -289,23 +278,55 @@ def visit_NamedInitializer(self, n):
     def visit_FuncDecl(self, n):
         return self._generate_type(n)
 
-    def _generate_struct_union(self, n, name):
-        """ Generates code for structs and unions. name should be either
-            'struct' or union.
+    def _generate_struct_union_enum(self, n, name):
+        """ Generates code for structs, unions, and enums. name should be
+            'struct', 'union', or 'enum'.
         """
+        if name in ('struct', 'union'):
+            members = n.decls
+            body_function = self._generate_struct_union_body
+        elif name in ('enum',):
```

Comment:
this should probably be:

```
else:
    assert name == 'enum'
    ... rest of code
```

Reply:
It is my understanding that `assert` is only proper in test code.  But, the gist that an exception should be raised for an invalid input does sound good.  How about leaving the `elif` and adding an `else` that raises a to-be-defined exception?

```
UnknownTypeError('`{name}` not recognized'.format(name=name))
```

Given that this is an internal method it seems that we may not need a nice user error.  I'll note that as is we'd get an `UnboundLocalError` for the following access to `members`.  I added a test for that and will adjust it based on the implementation decision.

Reply:
I actually think `assert` is fine in this case, since it's checking an invariant in the code - it's a bug if this place is reached, so `assert` is appropriate.